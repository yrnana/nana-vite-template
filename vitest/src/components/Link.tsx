import { useState } from 'react';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

type LinkProps = {
  page: string;
  children?: React.ReactNode;
};

const Link: React.FC<LinkProps> = ({ children, page }) => {
  const [status, setStatus] = useState(STATUS.NORMAL);

  const onMouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setStatus(STATUS.NORMAL);
  };

  return (
    <a
      className={status}
      href={page || '#'}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {children}
    </a>
  );
};

export default Link;
