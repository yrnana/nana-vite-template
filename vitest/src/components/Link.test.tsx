import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Link from './Link';

it('Link changes the class when hovered', () => {
  const { container } = render(<Link page="https://yrnana.dev">Yuri Na</Link>);

  expect(container).toMatchSnapshot('default');

  const link = screen.getByRole('link');

  userEvent.hover(link);
  expect(container).toMatchSnapshot('hover');

  userEvent.unhover(link);
  expect(container).toMatchSnapshot('unhover');
});
