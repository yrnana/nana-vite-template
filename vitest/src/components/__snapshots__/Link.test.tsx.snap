// Vitest Snapshot v1

exports[`Link changes the class when hovered > default 1`] = `
<div>
  <a
    class="normal"
    href="https://yrnana.dev"
  >
    Yuri Na
  </a>
</div>
`;

exports[`Link changes the class when hovered > hover 1`] = `
<div>
  <a
    class="hovered"
    href="https://yrnana.dev"
  >
    Yuri Na
  </a>
</div>
`;

exports[`Link changes the class when hovered > unhover 1`] = `
<div>
  <a
    class="normal"
    href="https://yrnana.dev"
  >
    Yuri Na
  </a>
</div>
`;
